bin/internal/global-validate.sh

```diff
@@ -0,0 +1,61 @@
+#!/bin/sh
+
+################################################################################
+# This program and the accompanying materials are made available under the terms of the
+# Eclipse Public License v2.0 which accompanies this distribution, and is available at
+# https://www.eclipse.org/legal/epl-v20.html
+#
+# SPDX-License-Identifier: EPL-2.0
+#
+# Copyright IBM Corporation 2020
+################################################################################
+
+################################################################################
+# This script will validate if the environment is well prepared to start Zowe.
+#
+# These environment variables should have already been loaded:
+# - INSTANCE_DIR
+#
+# Note: the INSTANCE_DIR can be predefined as global variable, or can be passed
+#       from command line "-c" parameter.
+################################################################################
+
+# if the user passes INSTANCE_DIR from command line parameter "-c"
+while getopts "c:" opt; do
+  case ${opt} in
+    c) INSTANCE_DIR=${OPTARG};;
+    \?)
+      echo "Invalid option: -${OPTARG}" >&2
+      exit 1
+      ;;
+  esac
+done
+
+########################################################
+# prepare environment variables
+export ROOT_DIR=$(cd $(dirname $0)/../../;pwd)
+. ${ROOT_DIR}/bin/internal/prepare-environment.sh -c "${INSTANCE_DIR}"
+
+########################################################
+if [[ "${USER}" == "IZUSVR" ]]
+then
+  echo "WARNING: You are running the Zowe process under user id IZUSVR. This is not recommended and may impact your z/OS MF server negatively."
+fi
+
+# source all utility libraries
+. ${ROOT_DIR}/bin/utils/utils.sh
+ERRORS_FOUND=0
+
+# Fix node.js piles up in IPC message queue
+# FIXME: where is the best place for this fix? Currently it's here because global-validate.sh script is supposed to run only once.
+. ${ROOT_DIR}/scripts/utils/cleanup-ipc-mq.sh
```

Comment:
Do you think we should move cleanup-ipc-mq.sh to bin/utils main library, or do we not want to consider it api?

Reply:
I always feel `cleanup-ipc-mq.sh` is not in the right place, but think it shouldn't be in utils and defined as api. This script is z/OS node.js specific and I feel it's a bug of node.js compiler. We can run it once globally and it doesn't need to be rerun by every node component. If we keep in global, maybe one day we can remove it easily once we drop support on the problematic compiler. Not sure if this makes sense to you.

Reply:
Yep, that's fine